Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+            timeStepIntervalTickColor = tintColor
+            thumbFillColor = UIColor.white
+            thumbBorderColor = tintColor
+            layerExtentFillColor = tintColor.withAlphaComponent(0.4)
+            pinnedThumbFillColor = tintColor
+            playbackButtonsFillColor = tintColor
+        }
+    }
+    
+    /*
+     A Boolean value indicating whether changes in the slider’s value
+     generate continuous update events. Default is True.
+     */
+    public var isContinuous: Bool = true
+    
+    // MARK: - Private Properties
+    
+    private var previousLocation = CGPoint()
+    
+    private var currentExtentStartTime: Date?
+    private var currentExtentEndTime: Date?
+    private var previousCurrentExtentStartTime: Date?
+    private var previousCurrentExtentEndTime: Date?
+    private var startTimeStepIndex: Int = -1
+    private var endTimeStepIndex: Int = -1
+    private var internalUpdate: Bool = false
+    
+    fileprivate var layerExtent: AGSTimeExtent?
+    
+    private let trackLayer = TimeSliderTrackLayer()
+    private let tickMarkLayer = TimeSliderTickMarkLayer()
+    private let lowerThumbLayer = TimeSliderThumbLayer()
+    private let upperThumbLayer = TimeSliderThumbLayer()
+    private let fullExtentStartTimeLabel: CATextLayer = CATextLayer()
+    private let fullExtentEndTimeLabel: CATextLayer = CATextLayer()
+    private let currentExtentStartTimeLabel: CATextLayer = CATextLayer()
+    private let currentExtentEndTimeLabel: CATextLayer = CATextLayer()
+    
+    private let maximumThumbSize: CGFloat = 50.0
+    private var trackLayerSidePadding: CGFloat = 30.0
+    private let labelSidePadding: CGFloat = 10.0
+    private let labelPadding: CGFloat = 3.0
+    private let paddingBetweenLabels: CGFloat = 5.0
+    
+    private var tickMarkLabels = [CATextLayer]()
+    fileprivate var tickMarks: [TickMark]?
+    
+    private var playPauseButton: UIButton?
+    private var forwardButton: UIButton?
+    private var backButton: UIButton?
+    
+    private var playImage: UIImage?
+    private var pauseImage: UIImage?
+    private var forwardImage: UIImage?
+    private var backImage: UIImage?
+    
+    fileprivate var pinnedThumbFillColor: UIColor = UIColor.black
+    
+    // If set to True, it will show two thumbs, otherwise only one. Default is True.
+    fileprivate var isRangeEnabled: Bool = true {
+        didSet {
+            upperThumbLayer.isHidden = !isRangeEnabled
+            currentExtentEndTimeLabel.isHidden = !isRangeEnabled
+        }
+    }
+    
+    private var heightConstraint: NSLayoutConstraint?
+    private var timer = Timer()
+    
+    private static var KVOContext = 0
+    private var map: AGSMap?
+    private var scene: AGSScene?
+    private var isObserving: Bool = false
+    
+    // MARK: - Override Functions
+    
+    required public init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
+    override public var frame: CGRect {
+        didSet {
```

Comment:
If I recall correctly, this property observer will not always be triggered by changes to the view's bounds, even though the frame would necessarily change. Are you wanting to redraw the view when the size changes? Consider setting the view's `contentMode` to `redraw`. If you need to perform an action when the view's size changes, a more reliable way is to query the `frame` in `layoutSubviews()`.

Reply:
Setting `contentMode` to `redraw` and `refresh` is already being called from `layoutSubviews()`.